src/OpenLoco/GameCommands/RenameCompanyOwner.cpp

```diff
@@ -0,0 +1,121 @@
+#include "../CompanyManager.h"
+#include "../Economy/Expenditures.h"
+#include "../Interop/Interop.hpp"
+#include "../Limits.h"
+#include "../Localisation/FormatArguments.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Localisation/StringManager.h"
+#include "../Types.hpp"
+#include "GameCommands.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    void sub_434BA1();
+
+    /**
+     * 0x00434914
+     * Renames a particular company owner.
+     *
+     * This command is called 3 times before the buffer is applied. Each time, 12 chars of the 36 char buffer are provided.
+     * The resulting company owner name has a maximum length of 31 chars; the last bytes are not used.
+     *
+     * @param flags @<bl> - game command flags
+     * @param companyId @<cl> - company id
+     * @param index @<ax> - update index (in order of: 1, 2, 0)
+     * @param buffer0 @<edx> - First part (4 chars) of the 12 update buffer
+     * @param buffer1 @<dx> - Second part (4 chars) of the 12 update buffer
+     * @param buffer2 @<bp> - Third part (4 chars) of the 12 update buffer
+     * @return @<ebx> - returns 0 if rename is successful; otherwise GameCommands::FAILURE
+     */
+    static uint32_t changeCompanyOwnerName(const uint8_t flags, CompanyId companyId, int16_t index, uint32_t buffer0, uint32_t buffer1, uint32_t buffer2)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Miscellaneous);
+
+        // Keep track of the company id over several calls.
+        static CompanyId _companyId{};
+        if (index == 1)
+            _companyId = companyId;
+
+        static uint32_t renameBuffer[9];
+
+        // Fill buffer over calls into the renameBuffer
+        if ((flags & GameCommands::Flags::apply) != 0)
+        {
+            static const std::array<int, 3> transformTable = { 2, 0, 1 };
+            int arrayIndex = transformTable.at(index);
+            renameBuffer[arrayIndex * 3] = buffer0;
+            renameBuffer[arrayIndex * 3 + 1] = buffer1;
+            renameBuffer[arrayIndex * 3 + 2] = buffer2;
+        }
+
+        // Applying the buffer?
+        if (index != 0)
```

Comment:
`index` doesn't change in this function, so is there a point to having this check all the way down here instead of the start of the function?

Reply:
nvm I understand now, we call this function 3 times with 3 inputs each ,and save them to the static buffer. madness